src/app/clusters/access-control-server/access-control-server.cpp

```diff
@@ -312,26 +312,33 @@ CHIP_ERROR AccessControlAttribute::WriteExtension(const ConcreteDataAttributePat
         }
         else if (count == 1)
         {
-            auto iterator = list.begin();
-            if (!iterator.Next())
+            bool processed     = false;
+            auto iterateStatus = list.Iterate([&](auto & item, bool &) -> CHIP_ERROR {
```

Comment:
Again, this is a breaking API change to a public API....

Reply:
It is, which I guess comes brings us to the question: if I can get the binary size to not increase, do you agree that generating a single type instead of a `DecodableType` + `Type` is worthwhile?

If not, then probably not worthwhile to continue the optimizations - but from a design/API perspective, I found that generating 2 types or 1 based on whether you have a list or not... is very confusing & clutters the API

Reply:
I agree that the current API is not amazing in terms of having to "use the right thing", but having a type that you can do some operations on but not others, depending on some internal state, is also not that great.... If someone gives you a DecodableList, can you do `begin()` on it?  Well, it depends...

If we can have a consistent API (which the Iterate thing certainly aims for) and not bloat codesize, I agree that it would be good to not have separate type representations for the two ends of the decode/encode pipeline.